validate/validate_test.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestGetMajorMinor(t *testing.T) {
+	cases := []struct {
+		version string
+		major   int
+		minor   int
+		err     error
+	}{
+		{"0.1beta", 0, 1, nil},
+		{"0.1.2", 0, 1, nil},
+		{"-1.-1beta", -1, -1, nil},
+		{"0.1", -1, -1, fmt.Errorf("have error")},
+		{"0", -1, -1, fmt.Errorf("have error")},
+		{"beta", -1, -1, fmt.Errorf("have error")},
+	}
+
+	for _, c := range cases {
+		ma, mi, e := getMajorMinor(c.version)
+		if (e != nil) != (c.err != nil) {
+			t.Errorf("Unexpected err, should %v, but got %v", c.err, e)
+		}
+		if ma != c.major {
+			t.Errorf("Unexpected major, should %v, but got %v", c.major, ma)
+		}
+		if mi != c.minor {
+			t.Errorf("Unexpected minor, should %v, but got %v", c.minor, mi)
+		}
+	}
+}
+
+func TestValidateKernelVersion(t *testing.T) {
+	const (
+		standardDesc = "Versions >= 2.6 are supported. 3.0+ are recommended"
+		errorDesc    = "Could not parse kernel version"
```

Comment:
I'd prefer not to duplicate these messages. Move these constants to validate.go, and use in both files.